src/librustdoc/html/static/main.js

```diff
@@ -2736,6 +2738,7 @@ function defocusSearchBar() {
                                      "",
                                      "?search=" + encodeURIComponent(search_input.value));
             }
+            document.title = searchTitle;
```

Comment:
When does `putBackSearch` run? Are you sure `search()` will be run before?

Reply:
`putBackSearch` is run when you focus the search input and it's not empty. Meaning that the search already ran at this point. ;)